.cloud-build/execute_changed_notebooks_helper.py

```diff
@@ -118,7 +124,15 @@ def select_notebook(changed_notebook: str,
         pass_count = 1
         fail_count = 0
 
-    return (random.randint(1, 100) * (1 + (fail_count / (pass_count + fail_count))) < test_percent)
+    inferred_failure_rate = fail_count / (pass_count + fail_count)
+
+    # If failure rate is high, the chance of testing should be higher
+    should_test_due_to_failure = random.uniform(0, 1) < inferred_failure_rate
+
+    # Additionally, only test a percentage of these
+    should_test_due_to_random_subset = random.uniform(0, 1) < test_percent
+
+    return should_test_due_to_failure or should_test_due_to_random_subset
```

Comment:
`should_test_due_to_failure and should_test_due_to_random_subset` makes more sense if you only want to test a subset of the testable notebooks.

Reply:
This code is only reachable when --test_percent < 100

Reply:
My comment is still relevant.

Reply:
I think that will require another --argument

Reply:
updated

Reply:
No, that’s not required.
I’m just asking you to change ‘or’ to ‘and’.

We should strive to reduce complexity/parameters where possible.

Reply:
done